app/controllers/posts_controller.rb

```diff
@@ -5,6 +5,28 @@ def index
   end
 
   def show
+    @user = User.find(params[:user_id])
     @post = Post.find(params[:id])
   end
+
+  def new
+    @user = User.find(params[:user_id])
+    @post = Post.new
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = @user.posts.new(post_params)
```

Comment:
- [ ] I would suggest here to make use of the current user that you have created on application controller, it is a good practice 

```
     @post = @current_user.posts.new(post_params)
```